base/broadcast.jl

```diff
@@ -363,4 +365,50 @@ for (sigA, sigB) in ((BitArray, BitArray),
     end
 end
 
+# additional definitions for Nullable
+# fast one & two-argument implementations
+function broadcast{T}(f, x::Nullable{T})
+    restype = promote_op(f, T)
+    if x.isnull
+        Nullable{restype}()
+    else
+        Nullable{restype}(f(x.value))
+    end
+end
+
+function broadcast(f, x::Union{Nullable, Number}, y::Union{Nullable, Number})
+    if isa(x, Number) && isa(y, Number)
+        throw(ArgumentError("broadcast is not supported on numbers"))
```

Comment:
I'd rather define a specialized method for two numbers, which could (and should) work. Currently it doesn't, so that would fix a bug with e.g. `broadcast(+, 1, 1)` (please add a test for that too).